Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift

```diff
@@ -732,7 +732,7 @@ class HTTPUpgradeTestCase: XCTestCase {
             let resultString = buffers.map { $0.getString(at: $0.readerIndex, length: $0.readableBytes)! }.joined(separator: "")
             assertResponseIs(response: resultString,
                              expectedResponseLine: "HTTP/1.1 101 Switching Protocols",
-                             expectedResponseHeaders: ["x-upgrade-complete: true", "upgrade: myproto", "connection: upgrade"])
+                             expectedResponseHeaders: ["X-Upgrade-Complete: true", "upgrade: myproto", "connection: upgrade"])
```

Comment:
Should we preserve emitted case, or should we explicitly keep our old behaviour?

Reply:
if we keep our old behaviour it will be more "expensive" and I think there is nothing that enforce us to do so... @weissi WDYT ?

Reply:
There's definitely nothing that forces us to do so, but I guarantee this will break a bunch of people's unit tests.

Reply:
We could add an optional flag to preserve case, and default it to `false`. That lets users have a fast path if they want it while preserving the current behaviour.

Reply:
@tanner0101 @weissi WDYT ? IMHO its fine as it is honestly as you can not depend on a case outcome (as any is impl specific).